packages/scripts/scripts/plugin-zip.js

```diff
@@ -47,10 +50,30 @@ if ( hasPackageProp( 'files' ) ) {
 	);
 }
 
+if ( noRootFolderArg !== undefined ) {
+	zipRootFolder = '';
+	stdout.write( 'Plugin files will be zipped without a root folder.\n\n' );
+} else if ( zipRootFolderArg !== undefined ) {
+	const trimmedZipRootFolderArg =
+		typeof zipRootFolderArg === 'string' ? zipRootFolderArg.trim() : null;
+	if ( trimmedZipRootFolderArg === null ) {
+		stdout.write(
+			'Please provide a `--root-folder` name or use `--no-root-folder.`\n\n'
+		);
+		process.exit( 1 );
+	}
+	zipRootFolder = `${ trimmedZipRootFolderArg }/`;
+	stdout.write(
+		`Adding the provided folder \`${ zipRootFolder }\` to the root of the package.\n\n`
```

Comment:
Nit: adding some spaces would make the output align better:

```suggestion
		`  Adding the provided folder \`${ zipRootFolder }\` to the root of the package.\n\n`
```

This is how it looks at the moment.

The default way to run the command:

<img width="878" alt="Screenshot 2024-10-28 at 08 57 48" src="https://github.com/user-attachments/assets/11e2b6ae-0060-47e0-9af5-c531c10f5ffb">

Custom folder name:

<img width="985" alt="Screenshot 2024-10-28 at 08 59 55" src="https://github.com/user-attachments/assets/42a57d4a-2ac0-430e-88ac-84480cf2e685">


No folder name:

<img width="985" alt="Screenshot 2024-10-28 at 08 59 55" src="https://github.com/user-attachments/assets/f2685231-94fc-4276-9cdf-37b5acf3f1d6">